libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFPasscodeKeyStore.m

```diff
@@ -80,21 +80,14 @@ - (SFKeyStoreKey *)keyStoreKey
         if (_keyStoreKey != nil)
             return _keyStoreKey;
         
-        NSString *keychainId = self.encryptionKeyKeychainIdentifier;
-        SFKeychainItemWrapper *keychainItem = [SFKeychainItemWrapper itemWithIdentifier:keychainId account:nil];
-        NSData *keyStoreKeyData = [keychainItem valueData];
-        if (keyStoreKeyData == nil) {
-            return nil;
-        } else {
-            NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc] initForReadingWithData:keyStoreKeyData];
-            _keyStoreKey = [unarchiver decodeObjectForKey:self.encryptionKeyDataArchiveKey];
-            [unarchiver finishDecoding];
-            
+        _keyStoreKey = [SFKeyStoreKey fromKeyChain:self.encryptionKeyKeychainIdentifier archiverKey:self.encryptionKeyDataArchiveKey];
```

Comment:
Should we deprecate this class in entirety and remove it in `8.0`?

Reply:
I should have marked it deprecated back in 6.0.  The problem is that no app is using that class directly, but not having it will impact whether you get a seamless migration when upgrading the SDK or not. We have never really stated the expected behavior when skipping a major release (or two).